### ISSUE TITLE
fsc overload resolution failure in .NET Core (coreclr)

### ISSUE THREAD

**Issue by enricosada** [Microsoft/visualfsharp#983]
I am tring to build [fsprojects/Argu](https://github.com/fsprojects/Argu/) in pr fsprojects/Argu#47 using .NET CLI with .NET Core target (`dnxcore50`) 

The `net46` version compile and run ok using the same fsc.exe in .NET CLI.

using `dnxcore50` target (.NET Core) instead fails

The [appveyor build log](https://ci.appveyor.com/project/nessos/argu-6jatx/build/1.0.20#L2272) show the failing build with multiple error who look similar

An example error:

```
C:\projects\argu-6jatx\src\Argu\ArgInfo.fs(229,64): error FS0495: The member or object constructor 'PreComputeUnionConstructor' has no argument or settable return property 'bindingFlags'. The required signature is static member FSharpValue.PreComputeUnionConstructor : unionCase:UnionCaseInfo * ?allowAccessToPrivateRepresentation:bool -> (obj [] -> obj). 
C:\projects\argu-6jatx\src\Argu\ArgInfo.fs(233,9): error FS0008: This runtime coercion or type test from type 
```

the `PreComputeUnionConstructor` it's trying to use the wrong overload.

the [ArgInfo.fs:229](https://github.com/fsprojects/Argu/blob/8da00cf8bf974a54e04cdbd6f4d1094f7619524c/src/Argu/ArgInfo.fs#L229) is

``` fsharp
let caseCtor = FSharpValue.PreComputeUnionConstructor(uci, bindingFlags = allBindings)
```

the `FSharpValue.PreComputeUnionConstructor` has multiple overload
### .NET CLI

.NET Command Line Tools (1.0.0-beta-001494)

Product Information:
 Version:     1.0.0-beta-001494
 Commit Sha:  c7a9d1d63a

Runtime Environment:
 OS Name:     Windows
 OS Version:  6.3.9600
 OS Platform: Windows
 Runtime Id:  win81-x64
### Packages
- Microsoft.FSharp.Core.netcore 1.0.0-alpha-151221. 
- Installing NETStandard.Library 1.0.0-rc2-23811

**Comment by KevinRansom** [comment]
Thank you for this.

**Comment by enricosada** [comment]
@KevinRansom or @dsyme any ideas for a workaround?
i updated to latest version ( `netstandard1.5` and latest lkg .net cli), same bug

to repro, just download the pr and use .net cli `1.0.0-beta-002071`

**Comment by KevinRansom** [comment]
I honestly haven’t had the time to even look at it.  I will try to check it out tomorrow if it is blocking, if it is not blocking and just a bug, can we let it lie for a bit?

Thanks

KEvin

**Comment by enricosada** [comment]
it's blocking `Argu` to build for .net core, for that i asked about a workaround.
but it's not a priority i think, thx @KevinRansom

**Comment by forki** [comment]
Argu is a dependency for other tools like paket. So I think it would be
beneficial if this could be solved soon. Thanks

**Comment by dsyme** [comment]
@enricosada  See also https://github.com/fsharp/FSharpLangDesign/blob/master/RFCs/FS-1013-enable-reflection-functionality-on-portable-profiles.md .  I think this may bee relevant here?

**Comment by ruxo** [comment]
Hi there, I also find this issue with `FuncConvert.ToFSharpFunc`.  For example:

```fsharp
    let h = FuncConvert.ToFSharpFunc<int,int>
```

This will cause following compilation error:

`error FS0502: The member or object constructor 'ToFSharpFunc' takes 1 type argument(s) but is here given 2. The required signature is 'static member FuncConvert.ToFSharpFunc : action:Action<'T> -> ('T -> unit)'`

I use compiler version `dotnet-compile-fsc/1.0.0-preview2-020000` with dotNET Core 1.1.

**Comment by dsyme** [comment]
My understanding is that this issue has been resoled and Argu was unblocked
